packages/functional-tests/src/app.ts

```diff
@@ -78,9 +84,59 @@ export default class App {
             const success = await test.run();
             console.log(`Test complete: '${testName}'. Success: ${success}`);
             this.rpc.send('functional-test:test-complete', testName, success);
-            this.rpc.send('functional-test:close-connection');
 
             delete this.activeTests[testName];
         }
     }
+
+
+    private async launchTestBrowser() {
+        while (true) {
+            const tester = MRESDK.Actor.CreateEmpty(this.context, {});
+            const selectedTestName = await new Promise<string>((resolve) => {

```

Comment:
Suggestion: Instead of an inline await on a new Promise, this could just be a call to an async method. Encapsulate the body of this promise into a method.

Reply:
done